src/components/TaskGenerator.js

```diff
@@ -1,32 +1,85 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import useBreakpoint, { MEDIA_QUERIES, MODE } from '../hooks/useBreakpoint';
+import { Category, Difficulty, Requirements } from './TaskTableCell';
+import { toggleCompleted } from '../store/tasks/tasks';
+import tasks from '../data/tasks';
 
 export default function TaskGenerator() {
-    const isLgViewport = useBreakpoint(MEDIA_QUERIES.LG, MODE.STRICT);
+    const [activeTask, setActiveTask] = useState(undefined);
+
+    const { tasks: reduxTasks } = useSelector(state => state.tasks);
+    const dispatch = useDispatch();
+    const isLgViewport = useBreakpoint(MEDIA_QUERIES.LG, MODE.GREATER);
+
+    const generateTask = () => {
+        const completedTaskIDsFromStore = Object.entries(reduxTasks)
+            .filter(([, taskObj]) => taskObj.completed)
+            .map(([taskId]) => +taskId);
+
+        let randomTask;
+
+        while (!randomTask) {
+            const generatedTask = tasks[Math.floor(Math.random() * tasks.length)];
+            if (!completedTaskIDsFromStore.includes(generatedTask.id)) randomTask = generatedTask;
+        }
+        setActiveTask(randomTask);
```

Comment:
Wonder if it makes more sense to filter down to uncompleted IDs and just grab one from that, rather than randomly rolling an ID and hoping it's one you haven't done. Could get a little hairy when you don't have many uncompleted tasks left. Plus I think it'll be easier to extend later to other filters (only ones with skill reqs, etc).

Maybe something like:
1. get array of keys from all tasks
2. get array of keys from completed tasks
3. take the difference of the two lists to find which ones we can select from

Reply:
adding on to that, we could also pretty easily exclude ignored tasks this way, since presumably you wouldn't want to ever get something you explicitly ignored

Reply:
Yeah makes sense, it's high chance of finding a random task when you have lots of uncompleted but could get messy later on. Will refactor.